providers/edge3/src/airflow/providers/edge3/plugins/www/src/components/MaintenanceExitButton.tsx

```diff
@@ -48,14 +51,43 @@ export const MaintenanceExitButton = ({ onExitMaintenance, workerName }: Mainten
   };
 
   return (
-    <IconButton
-      size="sm"
-      variant="ghost"
-      onClick={() => exitMaintenance()}
-      aria-label="Exit Maintenance"
-      title="Exit Maintenance"
-    >
-      <IoMdExit />
-    </IconButton>
+    <>
+      <IconButton
+        size="sm"
+        variant="ghost"
+        onClick={onOpen}
+        aria-label="Exit Maintenance"
+        title="Exit Maintenance"
+      >
+        <IoMdExit />
+      </IconButton>
+
+      <Dialog.Root onOpenChange={onClose} open={open} size="md">
+        <Portal>
+          <Dialog.Backdrop />
+          <Dialog.Positioner>
+            <Dialog.Content>
+              <Dialog.Header>
+                <Dialog.Title>Exit maintenance for worker {workerName}</Dialog.Title>
+              </Dialog.Header>
+              <Dialog.Body>
+                <p>Are you sure you want to exit maintenance mode for worker {workerName}?</p>
+              </Dialog.Body>
+              <Dialog.Footer>
+                <Dialog.ActionTrigger asChild>
+                  <Button variant="outline">No</Button>
+                </Dialog.ActionTrigger>
+                <Button onClick={exitMaintenance} colorScheme="red">
```

Comment:
For entering the maintenance we have no special color - should we change the other to red as well for consistency?

I assume you took over the code from delete button... which is more "dangerous". Not fully sure if we should mark it "red" as being dangerous or make both dialogs consistent?

Reply:
I don't think entering and removing from maintenance is particularly dangerous as such. I can make it red if you would like. However I wanted to reserve red color for the shutdown worker/remove worker buttons that I plan on adding next

Reply:
Then I propose _not_ making this red and taking "red" only for dangerous options like remote shutdown or deletion of worker.
```suggestion
                <Button onClick={exitMaintenance}>
```